pandas/tests/arrays/sparse/test_array.py

```diff
@@ -262,6 +262,18 @@ def test_take_negative(self):
         exp = SparseArray(np.take(self.arr_data, [-4, -3, -2]))
         tm.assert_sp_array_equal(self.arr.take([-4, -3, -2]), exp)
 
+    @pytest.mark.parametrize('fill_value', [0, None, np.nan])
+    def test_shift_fill_value(self, fill_value):
+        # GH #24128
+        sparse = SparseArray(np.array([1, 0, 0, 3, 0]),
+                             fill_value=8.0)
+        res = sparse.shift(1, fill_value=fill_value)
+        if fill_value is None:
```

Comment:
same here

Reply:
in this test it should be check for None only, the 0 and np.nan have their own behavior

Reply:
I replaced eventually, it uses isna everywhere now